src/test/java/org/openrewrite/java/migrate/guava/NoGuavaMapsNewHashMapTest.java

```diff
@@ -0,0 +1,120 @@
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.DocumentExample;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+
+class NoGuavaMapsNewHashMapTest implements RewriteTest {
+
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec
+          .recipe(new NoGuavaMapsNewHashMap())
+          .parser(JavaParser.fromJavaVersion().classpath("guava"));
+    }
+
+    @DocumentExample
+    @Test
+    void replaceWithNewHashMap() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+              import com.google.common.collect.*;
+                            
+              import java.util.Map;
+                            
+              class Test {
+                  Map<Integer, Integer> cardinalsWorldSeries = Maps.newHashMap();
+              }
+              """,
+            """
+              import java.util.HashMap;
+              import java.util.Map;
+                            
+              class Test {
+                  Map<Integer, Integer> cardinalsWorldSeries = new HashMap<>();
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void replaceWithNewHashMapWithMap() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+              import com.google.common.collect.*;
+                            
+              import java.util.Collections;
+              import java.util.Map;
+                            
+              class Test {
+                  Map<Integer, Integer> m = Collections.emptyMap();
+                  Map<Integer, Integer> cardinalsWorldSeries = Maps.newHashMap(m);
+              }
+              """,
+            """
+              import java.util.Collections;
+              import java.util.HashMap;
+              import java.util.Map;
+                            
+              class Test {
+                  Map<Integer, Integer> m = Collections.emptyMap();
+                  Map<Integer, Integer> cardinalsWorldSeries = new HashMap<>(m);
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void replaceWithNewHashMapWithCapacity() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+              import com.google.common.collect.*;
+                            
+              import java.util.HashMap;
+              import java.util.Map;
+                            
+              class Test {
+                  Map<Integer, Integer> cardinalsWorldSeries = Maps.newHashMapWithExpectedSize(2);
+              }
+              """,
+            """
+              import java.util.HashMap;
+              import java.util.Map;
+                            
+              class Test {
+                  Map<Integer, Integer> cardinalsWorldSeries = new HashMap<>(2);
```

Comment:
I _think_ the recommended form here is `Hashmap.newHashMap(capacity)` to get the immediately right sized collection instead of repeated allocations. Might be a good recipe in and of itself.

Reply:
You are right, I removed the `newHashMapWithExpectedSize` from being replaced.
Probably  `newLinkedHashMapWithExpectedSize` should also be excluded from [NoGuavaMapsNewLinkedHashMap](https://github.com/openrewrite/rewrite-migrate-java/blob/9c6acfb5c5f2c942f8660a687b5263b3841c65ac/src/main/java/org/openrewrite/java/migrate/guava/NoGuavaMapsNewLinkedHashMap.java#L34) aswell (see [google/common/collect/Maps.html](https://guava.dev/releases/19.0/api/docs/com/google/common/collect/Maps.html#newLinkedHashMapWithExpectedSize(int)))?

Reply:
Might be good to be consistent indeed I, thanks for calling that out!